---
title: "bact_visualization_viz"
author: "Sam"
date: "2022-11-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

---

```{r}
#set user defined path
setwd()
file_path <- "path/to/bacteria"

```

```{r}
#loading the required library
library(phyloseq)
library(ggplot2)
```

```{r}
#rm(list=ls())
#loading the ASV taxonomic table
tax<-read.table("data/ASV_tax_species.tsv", header = T, row.names = 1,sep = "\t")
head(tax)
names(tax)
tax<-tax[,-c(9:11)]
#head(tax)
tax<-t(tax)
tax<-as.matrix(tax) #converting tax dataframe to a matrix
#head(tax)
tax<-(t(tax)) 
tax<-tax_table(tax) #converting the tax matrix to phyloseq object
#class(tax)
#head(tax)
```

```{r}

# loading the ASv_counts tables
otu<-read.table("data/ASV_table.tsv", header = T, row.names=1,sep="\t")
otu<-otu[,-c(21)]
head(otu)
names(otu)

otu<-otu_table(otu,taxa_are_rows = TRUE) #converting otu matrix to phyloseq object.
head(otu)
```

```{r}
#combine the two physeq object i.e tax,otu

physeq_b <-phyloseq(otu, tax)
physeq_b
#plot_bar(physeq, fill = "Class")
```

```{r}
#importing the metadata
meta<- read.table("data/bact_meta",sep = "\t", header = TRUE, row.names = 1)
head(meta)
#names(meta) = c("Sample_ID","Animal_Type",  "Microbiome.Domain","Location")
names(meta)
head(meta)
meta<-sample_data(meta) #converting the metadata to a phyloseq object
```

```{r}
# we now create a third object called random for mergingwith the other three object
library("ape")
random_tree <- rtree(ntaxa(physeq), rooted = TRUE, tip.label = taxa_names(physeq))
plot(random_tree)

```

```{r}
#merging the preceeding 3 objects.
physeq1 <- merge_phyloseq(physeq, meta, random_tree)
physeq1
saveRDS(physeq1, file="physeq_rumenbacteria.rds")
#readRDS(file ="path/to/physeq_rumenbacteria.rds" ) #reading a .rds image
```
# Some packages above have conficts in libraries change to a different R session before proceeding down

```{r}
#Load Packages
suppressPackageStartupMessages({
require(csv)
require(dplyr)
require(tidyr)
library(dada2)
library(Cairo)
library(ggtree)
library(VennDiagram)
library(UpSetR)
library("phyloseq"); packageVersion("phyloseq") # Handling and analysis of high-throughput microbiome census data.
library("vegan");packageVersion("vegan") # Community Ecology Package.
library("ggplot2");packageVersion("ggplot2") # Create Elegant Data Visualisations Using the Grammar of Graphics.
library("dendextend");packageVersion("dendextend")
library("tidyr");packageVersion("tidyr")
library("viridis");packageVersion("viridis")
library("reshape");packageVersion("reshape")
#install.packages("remotes")
#remotes::install_github("vmikk/metagMisc")
library(metagMisc)
library(janitor)
library(coin) # Conditional Inference Procedures in a Permutation Test Framework.
library(reshape2) # Flexibly Reshape Data: A Reboot of the Reshape Package. 
library(ggnewscale) # Multiple Fill and Colour Scales in 'ggplot2'.
library(MicrobiotaProcess) # an R package for analysis, visualization and biomarker discovery of Microbiome.
library(patchwork)
})
```

```{r}

#Read in Data
ps1 <- readRDS(file.path(file_path, "rds/physeq_bacteria.rds"))

```

```{r}
#filtering the unwanted sequences
ps2 <- subset_taxa(ps1, (Order!="Chloroplast") | is.na(Order))
ntaxa(ps2)
ps2 <- subset_taxa(ps2, (Genus!="Chloroflexi") | is.na(Genus))
ntaxa(ps2)
ps2<- subset_taxa(ps2, (Family!="Mitochondria") | is.na(Family))
ntaxa(ps2)
ps2<- subset_taxa(ps2, (Kingdom!="Archaea") | is.na(Kingdom))
ntaxa(ps2)
ps2<- subset_taxa(ps2, (Kingdom!="Eukaryota") | is.na(Kingdom))
ntaxa(ps2)

```

```{r}
ps3 <- prune_taxa(taxa_sums(ps2) > 20, ps2)
ps3

```

```{r}

#Extracting the filtered taxonomy and feature tables for barplot plotting
tax_table <- phyloseq_to_df(ps3, addtax = T, addtot = F, addmaxrank = F)

cumulation <- tax_table %>% adorn_totals(c("col"))
cumulation <- cumulation[order(cumulation$Total, decreasing = TRUE),]


```

```{r}
#merging the blast taxonomic classification to blast abundance table
merged_data <- tax_table
write.csv(merged_data, file="data/Taxonomic_Classification.csv")
```

```{r}

#grouping the data (entire dataset): Genus, Species and sample names
#Featured_table <- merged_data[,c(8:9,10:29)] #You remain with genus and species

Featured_table <- merged_data[,c(3:9,10:29)]#Get the ranks from kingdom to species
group <- Featured_table %>%
  group_by(Genus)%>%
  summarise_if(is.numeric, sum)

#Groups the data in the defined order which eases downstream analysis

group <- Featured_table %>% group_by(Genus)%>%
  summarise_each(funs(sum), "C1","C2","C3","C4","C5","CA1","CA2","CA3","CA4","CA5",
                 "G1","G2","G3","G4","G5", "S1","S2","S3","S4","S5")

group<-group[-c(1),]
View (group)

```

```{r}
#creating multiple dataframes for the different treatments
cow<-group[,c(1,2:6)]
camel<-group[,c(1,7:11)]
goat<-group[,c(1,12:16)]
sheep<-group[,c(1,17:21)]

all<-group[,c(1,2:21)]
#region<-group[]
```



```{r}
cow_total <- cow %>% adorn_totals(c("col"))
cow_total <- mutate(cow_total, cow=rowSums(cow_total[6])/5)
cow_total <- cow_total[,c(1,7)]

camel_total <- camel %>% adorn_totals(c("col"))
camel_total <- mutate(camel_total, camel=rowSums(camel_total[6])/5)
camel_total <- camel_total[,c(1,7)]

goat_total <- goat %>% adorn_totals(c("col"))
goat_total <- mutate(goat_total, goat=rowSums(goat_total[6])/5)
goat_total <- goat_total[,c(1,7)]

sheep_total <- sheep %>% adorn_totals(c("col"))
sheep_total <- mutate(sheep_total, sheep=rowSums(sheep_total[6])/5)
sheep_total <- sheep_total[,c(1,7)]

all_total <- all %>% adorn_totals(c("col"))
all_total <- mutate(all_total, all=rowSums(all_total[21])/20)
all_total <- all_total[,c(1,22)]

```

```{r}

#merging the above dataframes
merged <- Reduce(function(x,y) merge(x,y,by="Genus",all=TRUE),
list(cow_total, camel_total, goat_total, sheep_total,all_total))

names(merged)<-c('Genus','Cow','Camel','Goat','Sheep','All')
#calculating the total abundance per genus and ordering from the most abundant to the lowest
cumulation <- merged %>% adorn_totals(c("col"))
cumulation <- cumulation[order(cumulation$Total, decreasing = TRUE),]
cumulation$perc = cumulation$Total / sum(cumulation$Total) * 100

```

```{r}
#Original_for Figure 1_specifying the taxa to be tabulated
head(cumulation, n=15)
head(cumulation$Genus, n=14) # Top 14 genus

to_represent <-c("Rikenellaceae RC9 gut group","Christensenellaceae R-7 group","Prevotella",
                 "NK4A214 group","Saccharofermentans","Butyrivibrio","Psychrobacter",
                 "Lachnospiraceae XPB1014 group","Candidatus  Saccharimonas","Succiniclasticum",
                 "Prevotellaceae UCG-003", "Prevotellaceae UCG-001", 
                 "[Ruminococcus] gauvreauii group","Lachnospiraceae AC2044 group", "Others")

#to_represent<-c("Firmicutes", "Bacteroidota","Proteobacteria", "Patescibacteria", "Spirochaetota", "Planctomycetota", "Verrucomicrobiota", "Actinobacteriota", "Desulfobacterota", "Synergistota", "Armatimonadota", "Fibrobacterota", "SAR324 clade(Marine group B)", "Bdellovibrionota", "Others")

#aggregating the rest of the phyla as others
#grouped_data <- aggregate(merged[-1], list(Genus = replace(merged$Genus,!(merged$Genus %in% to_represent), "Others")), sum)
```

```{r}
grouped_data <- aggregate(merged[-1], list(Genus = replace(merged$Genus,!(merged$Genus %in% to_represent), "Others")), sum)
View(grouped_data) 


#regions_d<-grouped_data[,c(1,2:10)]
#diet<-grouped_data[,c(1,11:13)]
#regions<-grouped_data[,c(1,14:16)]
#all

cow<-grouped_data[,c(1:2)]
camel<-grouped_data[,c(1,3)]
goat<-grouped_data[,c(1,4)]
sheep<-grouped_data[,c(1,5)]


all<-grouped_data[,c(1:5)]
#w <-grouped_data[,c(1:6)]
#order<- w[order(w$All, decreasing = TRUE), ]
#all1 <- order[,c(1:5)]
```

## Viewing All samples Diversity

```{r}
#install.packages("janitor")
library(janitor)
#converting the abudances into percentage
bar_all <- adorn_percentages(all, denominator = "col", na.rm = FALSE)
#barp <- adorn_percentages(merged, denominator = "col", na.rm = FALSE)

#2-way tabyls % - Yosef added
bar_all %>%
  adorn_totals("row") %>%
  adorn_pct_formatting()

dist_all<-bar_all %>%
  adorn_totals("row") %>%
  adorn_pct_formatting()
write.csv(dist_all,"C:/Sam/bacteria/data/All_Diversity_bacteria.csv")

#gathering the data
bar_all <- bar_all %>%
  gather(value = "abundance", key = "Colonization_status", -Genus)
bar_all <- as.data.frame(gsub("\\(", " (", as.matrix(bar_all)))


# coerce the dataframe columns into respective data type
bar_all$Genus <- as.factor(bar_all$Genus)
bar_all$Colonization_status <- as.character(bar_all$Colonization_status)
bar_all$abundance <- as.numeric(bar_all$abundance)

#ordering the data for plotting
bar_all$Genus <- reorder(bar_all$Genus, bar_all$abundance)
bar_all$Genus <- factor(bar_all$Genus, levels=rev(levels(bar_all$Genus)))
bar_all$Genus <- factor(bar_all$Genus,
                        levels=c("Rikenellaceae RC9 gut group","Christensenellaceae R-7 group","Prevotella",
                "NK4A214 group","Saccharofermentans","Butyrivibrio","Psychrobacter",
                "Lachnospiraceae XPB1014 group","Candidatus  Saccharimonas","Succiniclasticum",
                 "Prevotellaceae UCG-003", "Prevotellaceae UCG-001", 
                "[Ruminococcus] gauvreauii group","Lachnospiraceae AC2044 group", "Others"))

# Choosing the colours to use in the barplot
myPalette <-c("#04bf17","blue","purple", "#f44336",  "#5e4f20", "#ef419c", "#5591c7", "#74c755", "#efba16", "#511ff2", "#8c9a85", "#412333", "#ea9999", "#63e9f7","#0c343d", "#bf9000", "#783f04") 

 # c("#DA5724", "#74D944", "#CE50CA", "#3F4921","#89C5DA", "#C0717C", "#CBD588", "#5F7FC7", "#673770", "#D3D93E", "#38333E", "#508578", "#D7C1B1", "#689030", "#AD6F3B")

```
```{r}
# change the relevant label to "italic non-italic"
lbs = brk = levels(bar_all$Genus)
lbs[match("Rikenellaceae RC9 gut group", brk)] = expression(italic("Rikenellaceae RC9 gut group"))
lbs[match("Christensenellaceae R-7 group", brk)] = expression(italic("Christensenellaceae R-7 group"))
lbs[match("Prevotella", brk)] = expression(italic("Prevotella"))
lbs[match("NK4A214 group", brk)] = expression(italic("NK4A214 group"))
lbs[match("Saccharofermentans", brk)] = expression(italic("Saccharofermentans"))
lbs[match("Butyrivibrio", brk)] = expression(italic("Butyrivibrio"))
lbs[match("Psychrobacter", brk)] = expression(italic("Psychrobacter"))
lbs[match("Lachnospiraceae XPB1014 group", brk)] = expression(italic("Lachnospiraceae XPB1014 group"))
lbs[match("Candidatus  Saccharimonas", brk)] = expression(italic("Candidatus Saccharimonas"))
lbs[match("Succiniclasticum", brk)] = expression(italic("Succiniclasticum"))
lbs[match("Prevotellaceae UCG-003", brk)] = expression(italic("Prevotellaceae UCG-003"))
lbs[match("Prevotellaceae UCG-001", brk)] = expression(italic("Prevotellaceae UCG-001"))
lbs[match("[Ruminococcus] gauvreauii group", brk)] = expression(italic("Ruminococcus gauvreauii group"))
lbs[match("Lachnospiraceae AC2044 group", brk)] = expression(italic("Lachnospiraceae AC2044 group"))
lbs[match("Others", brk)] = expression(plain("Others"))
```

```{r}
#bar_all <- read.csv("bar_all.csv", sep = ",", header = T)
#bar_all <- bar_all[,-1]
#attach(bar_all)
#View(bar_all)

library(Cairo)
library(forcats)
#plotting the barplot 
p_all <- ggplot(bar_all,aes(x = fct_inorder(Colonization_status), y = abundance), labs(fill= Genus), group=row.names(bar_all))+ xlab("Colonization_status")+ ylab("abundance") + geom_col(aes(fill = Genus),position = position_stack(reverse = FALSE))+
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic", family = "Arial"))+
  scale_fill_manual(values = myPalette,labels = lbs, breaks = brk)+
  guides(fill = guide_legend(reverse = FALSE))+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic", family = "Arial"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10, family = "Arial"))+
  theme(axis.text = element_text(colour = "black", size = 10, family = "Arial"))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.15, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.position = "right", legend.justification = "top", legend.direction = "vertical", legend.text = element_text(size = 10))+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10, family = "Arial"))+
  theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10, family = "Arial")+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90, family = "Arial"))+
          theme(axis.title.x = element_text(size = 10, angle = 0, family = "Arial")))

#install.packages('extrafont')
library('forcats')
```

```{r}
library(extrafont)
#font_import()

abundance_bact<-p_all + scale_y_continuous(labels = scales::percent_format(accuracy = 1)) + xlab("Animal") + theme(legend.text = element_text(size = 10))
  
abundance_bact


```

```{r}
ggsave(file.path(file_path, "results/abundance/abundance_bacteria.png"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/abundance/abundance_bacteria.jpeg"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/abundance/abundance_bacteria.svg"), width=12, height=12, dpi=600)

```


```{r}
library(ggpubr)

```

### BETA DIVERSITY

```{r}
#weighted unifrac
#Palette <- c('#89C5DA', "#DA5724", "#74D944", "#CE50CA", "#3F4921", "#C0717C", "#CBD588", "#5F7FC7", "#673770", "#D3D93E", "#38333E", "#508578", "#D7C1B1", "#689030", "#AD6F3B", "#CD9BCD", "#D14285", "#6DDE88", # "#652926", "#7FDCC0", "#C84248", "#8569D5", # "#5E738F", "#D1A33D", "#8A7C64", "#599861")

Palette <- c("#6f677f", "#375d8e", "#000403", "#065411")           

ordu = ordinate(ps3, "PCoA", "unifrac", weighted = TRUE)
p <- plot_ordination(ps3, ordu, color="Animal_Type")+ geom_point(size=2) +
  scale_color_manual(values = Palette) + 
  theme(axis.text.x = element_text(angle = 0, hjust = 1))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.text = element_text(size = rel(1), colour = "black"))+
  theme(legend.title = element_text(face = NULL))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = rel(1)))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))

beta_bact <-p + stat_ellipse()  + labs(tag = "B")

```


```{r}
ggsave( "D:/merid/bacteria/beta_bact.png", 
       width = 12, height = 12, dpi = 600)

ggsave("D:/merid/bacteria/beta_bact.jpeg",
       width = 12, height = 12, dpi = 600)

ggsave( "D:/merid/bacteria/beta_bact.svg", 
       width = 12, height = 12, dpi = 600)

```

### ALPHA DIVERSITY

```{r}
###ALPHA PART TWO
#alpha diversity estimation
physeq4 <- ps3
physeq4

#checking out the total read counts in the samples
reads <- sample_sums(physeq4)
reads

summary(sample_sums(physeq4))

library(microbiome)
#Extracting the otu table from the phyloseq object and plotting the rarefaction curve
otu_tab <- t(abundances(physeq4))
p <- vegan::rarecurve(otu_tab, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab), 
                                   col = "blue", cex = 0.6))

set.seed(9242)  

#calculatin an even sampling depth for all the samples
rarefied_bact <- rarefy_even_depth(physeq4, sample.size = 1632)
rarefied_bact <- rarefy_even_depth(physeq4, sample.size = 41)
rarefied_bact
```

```{r}
library(tidyverse)
#calculating the alpha diversity
diversity <- alpha(rarefied_bact, index = "all")
diversity <- rownames_to_column(diversity, "sample_id")

#Extracting the sample metadata from the phyloseq object
sdata1 <- meta(physeq4)
sdata1 <- rownames_to_column(sdata1, "sample_id")

#Extracting the shannon diversity index
shannon <- diversity %>% select(sample_id, diversity_shannon)

shannon_edited <- merge(shannon, sdata1, by = "sample_id", all = TRUE)
#shannon_edited <- shannon_edited[c(1:18, 62:86, 19:61),]

#confirming if the shannon indices are normally distributed
shapiro.test(shannon_edited$diversity_shannon)
shannon_edited$diversity_shannon<-log1p(shannon_edited$diversity_shannon)
library(ggpubr)

#plotting the boxplots for the shannon index data
Pa <- ggboxplot(shannon_edited, "Animal_Type","diversity_shannon",
               color = "Animal_Type", palette = c("#6f677f", "#375d8e", "#000403", "#065411"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + #+ stat_compare_means()+
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))#+
  #stat_compare_means()
bact_shannon <-Pa + aes(x = fct_inorder(Animal_Type)) + theme(legend.position = "none") + xlab("Animal Type") + ylab("Shannon diversity")

bact_shannon

```
```{r}
ggsave(file.path(file_path, "results/alpha/bact_shannon.png"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_shannon.jpeg"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_shannon.svg"), width=12, height=12, dpi=600)
```

```{r}
#Chao1 diversity estimates
chao1 <- diversity %>% select(sample_id, chao1)
chao_edited <- merge(chao1, sdata1, by = "sample_id", all = TRUE)
#chao_edited <- chao_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(chao_edited$chao1)

#plotting chao1 distribution boxplot
bact_chao1 <- ggboxplot(chao_edited, "Animal_Type","chao1",
               color = "Animal_Type", palette = c('#89C5DA', "#DA5724", "#74D944", "#CE50CA"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + 
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0))) 
bact_chao1 <- bact_chao1  + aes(x = fct_inorder(Animal_Type)) + theme(legend.position = "none") + xlab("Animal Type") 

bact_chao1
```
```{r}
ggsave(file.path(file_path, "results/alpha/bact_chao1.png"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_chao1.jpeg"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_chao1.svg"), width=12, height=12, dpi=600)
```

```{r}
#Evenness diversity estimates
evenness_pielou <- diversity %>% select(sample_id, evenness_pielou)
evenness_pielou_edited <- merge(evenness_pielou, sdata1, by = "sample_id", all = TRUE)
#evenness_pielou_edited <- evenness_pielou_edited[c(1:18, 62:86, 19:61),]

#confirming if the chao1 indices are normally distributed
shapiro.test(evenness_pielou_edited$evenness_pielou)

#plotting chao1 distribution boxplot
bact_pielou <- ggboxplot(evenness_pielou_edited, "Animal_Type","evenness_pielou",
               color = "Animal_Type", palette = c("#38333E", "#508578", "#D7C1B1", "#689030"),
               add = "jitter", linetype = "solid", Family = "Palatino Linotype", add.params = list(),
               error.plot = "pointrange", legand = NULL, size = NULL, width = 0.7, notch = FALSE, outlier.shape = 20, facet.by = NULL,
               panel.labs = NULL, short.panel.labs = TRUE,bxp.errorbar = FALSE, bxp.errorbar.width = 0.4, ggtheme = theme_pubr())+
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +  
  theme(legend.text = element_text(size = 10, colour = "black", face = "italic"), legend.text.align = 0)+
  theme(axis.text.y = element_text(angle = 0, vjust = 0.5, size = 10))+
  theme(axis.text = element_text(colour = "black", size = 10))+
  theme(axis.line = element_line())+
  theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = NULL, size = 1))+
  theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
  theme(legend.justification = "top")+
  theme(legend.position = "right")+
  theme(legend.key = element_rect(fill = "white"))+
  theme(legend.title = element_text(face = NULL, size = 10))+theme(panel.background = element_blank(), axis.text = element_blank())+
  theme(axis.text = element_text(colour = "black", size = 10)+
          theme(axis.line = element_line())+
          theme(panel.background = element_rect(fill = "white"),plot.margin = margin(0.5, 0.5, 0.5, 0.5, "cm"), plot.background = element_rect(colour = "grey"))+
          theme(axis.ticks.length.y = unit(.25, "cm"), axis.ticks.length.x = unit(.25, "cm"), axis.text.x = element_text(margin = margin(t = .3, unit = "cm")))+
          theme(axis.title.y = element_text(size = 10, face = "plain", angle = 90))+
          theme(axis.title.x = element_text(size = 10, angle = 0)))
bact_pielou <- bact_pielou + aes(x = fct_inorder(Animal_Type)) + theme(legend.position = "none") + xlab("Animal Type") + ylab("Evenness diversity")

bact_pielou

```
```{r}
ggsave(file.path(file_path, "results/alpha/bact_pielou.png"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_pielou.jpeg"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_pielou.svg"), width=12, height=12, dpi=600)
```

### merge
```{r}
require(ggpubr)
require(ggplot2)
require(svglite)

bact_shannon
bact_chao1
bact_pielou

alpha_bact <- ggarrange(bact_shannon, bact_chao1, bact_pielou, labels = c("D", "E", "F"), nrow = 1)
```
```{r}
ggsave(file.path(file_path, "results/alpha/bact_alpha.png"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_alpha.jpeg"), width=12, height=12, dpi=600)
ggsave(file.path(file_path, "results/alpha/bact_alpha.svg"), width=12, height=12, dpi=600)
```


##  Permutational Multivariate Analysis of Variance
```{r}
distme <- get_dist(ps3, distmethod ="bray", method="hellinger")
sampleda <- data.frame(sample_data(ps3), check.names=FALSE)
sampleda <- sampleda[match(colnames(as.matrix(distme)),rownames(sampleda)),,drop=FALSE]
sampleda$Animal_Type <- factor(sampleda$Animal_Type)
set.seed(1024)
adores <- adonis(distme ~ Animal_Type, data=sampleda, permutation=9999)
data.frame(adores$aov.tab)


```

## Biomarker discovery

```{r}
deres <- diff_analysis(obj = ps3, classgroup = "Animal_Type",
                       mlfun = "lda",
                       filtermod = "pvalue",
                       firstcomfun = "kruskal_test",
                       firstalpha = 0.05,
                       strictmod = TRUE,
                       secondcomfun = "wilcox_test",
                       subclmin = 3,
                       subclwilc = TRUE,
                       secondalpha = 0.01,
                       lda=3)
deres
```


```{r}
diffclade_bact <- ggdiffclade(
                   obj=deres, 
                   alpha=0.3, 
                   linewd=0.15,
                   skpointsize=0.6, 
                   layout="radial",
                   taxlevel=7, 
                   removeUnkown=TRUE,
                   reduce=TRUE # This argument is to remove the branch of unknown taxonomy.
               ) +
               scale_fill_manual(
                   values=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA")
               ) +
               guides(color = guide_legend(
                                  keywidth = 0.1, 
                                  keyheight = 0.6,
                                  order = 3,
                                  ncol=1)
               ) +
               theme(
                   panel.background=element_rect(fill=NA),
                   legend.position="right", 
                   plot.margin=margin(0,0,0,0),
                   legend.spacing.y=unit(0.02, "cm"), 
                   legend.title=element_text(size=7),
                   legend.text=element_text(size=6), 
                   legend.box.spacing=unit(0.02,"cm")
               )  + theme(legend.text = element_text(size = 10))
diffclade_bact

```

## Venn Diagram

```{r}
#vennlist <- get_vennlist(obj=ps1, factorNames="Animal_Type")
vennlist <- get_vennlist(obj=ps3, factorNames="Animal_Type")
venn_bact <- venn.diagram(vennlist,
                      height=5,
                      width=5, 
                      filename=NULL, 
                      fill=c("#89C5DA", "#DA5724", "#74D944", "#CE50CA"),
                      cat.col=c("#508578", "#D7C1B1", "#689030", "#AD6F3B"),
                      alpha = 0.85, 
                      fontfamily = "serif",
                      fontface = "bold",
                      cex = 1.2,
                      cat.cex = 1.3,
                      cat.default.pos = "outer",
                      cat.dist=0.1,
                      margin = 0.1, 
                      #lwd = 3,
                      #lty ='dotted',
                      lwd = 7,
                      lty ='dotted',
                      imagetype = "svg", output=TRUE ) 
grid::grid.draw(venn_bact)
```

```{r}
library(phyloseq)
library(VennDiagram)
library(grid)
library(dplyr)

# Assuming 'ps1' is a phyloseq object

# Aggregating the phyloseq object at the genus level
ps1_genus <- tax_glom(ps3, "Genus")

# Creating the Venn list using the genus-level data
vennlist <- get_vennlist(obj = ps1_genus, factorNames = "Animal_Type")

# Plotting the Venn diagram
venn_genus <- venn.diagram(vennlist,
                           height = 5,
                           width = 5,
                           filename = NULL,
                           fill=c("#511ff2","#63e9f7","#689030","#FFAC0A"),
                      cat.col=c(  "#DA5724","#508578","#e2ca08" , "#AD6F3B"),
                           alpha = 0.85,
                           fontfamily = "serif",
                           fontface = "bold",
                           cex = 1.2,
                           cat.cex = 1.3,
                           cat.default.pos = "outer",
                           cat.dist = 0.1,
                           margin = 0.1,
                           lwd = 3,
                           lty = 'dotted',
                           imagetype = "jpeg",
                           output = TRUE)

grid::grid.draw(venn_genus)

```
```{r}
# Generate a color palette of 142 colors
palette <- colors()[1:142]

# Convert color codes to quoted strings
palette_quoted <- sprintf("\"%s\"", palette)

# Join color codes with commas
palette_formatted <- paste(palette_quoted, collapse = ", ")

# Print the formatted color palette
cat(palette_formatted)

```


